src/components/core/Header.vue

```diff
@@ -24,83 +43,279 @@ export default {
       flat,
       round,
-      size='md',
-      icon='wallet',
-      color='grey-6',
-      :to='{ name: "wallet" }'
+      no-wrap,
+      icon='account_circle',
+      label='Account',
+      v-if='isLoggedIn'
     )
-    //- q-btn(v-if="isLoggedIn" @click="logout") Logout
-    //- q-btn(v-if="!isLoggedIn" to="login" flat) Login
-    //- q-btn(v-if="!isLoggedIn" to="register" outline) Register
-    div(v-if='$q.screen.lt.md')
-      q-btn(flat, round, v-if='!isLoggedIn', dense, :to='{ name: "login" }')
-        q-icon.material-icons-outlined(name='account_circle')
-    div(v-if='$q.screen.gt.sm')
-      q-btn(dense, outline, v-if='!isLoggedIn', :to='{ name: "login" }')
-        q-avatar.q-mr-sm(rounded, size='30px')
-          q-icon.material-icons-outlined(name='account_circle')
-        .q-mr-sm Sign In
-    q-btn(dense, flat, round, size='md', no-wrap, v-if='isLoggedIn')
-      // TODO Change avatar when logged in
-      q-avatar(rounded, size='30px')
-        q-icon.material-icons-outlined(name='account_circle', color='grey-6')
-      //- q-icon(name='arrow_drop_down' size='16px')
-      q-menu(auto-close, fit)
+      q-menu.menu-edit(auto-close, fit)
         q-list(dense)
+          q-item.menu-link(
+            v-if='isLoggedIn',
+            clickable,
+            :to='{ name: "profile", params: { profile: 1 } }'
+          )
+            q-item-section.q-pa-sm
+              q-btn(
+                flat,
+                icon='account_circle',
+                label='Profile',
+                dense,
+                align='left',
+                font-size='10px'
+              )
+          q-separator(v-if='isLoggedIn')
+          // TODO add params for profile, gallery and wallet
+          q-item.menu-link(clickable, :to='{ name: "wallet" }')
+            q-item-section.q-pa-sm
+              q-btn(
+                flat,
+                icon='wallet',
+                label='Wallet',
+                dense,
+                align='left',
+                font-size='10px'
+              )
+          q-separator(v-if='isLoggedIn')
+          q-item.menu-link(
+            clickable,
+            :to='{ name: "account-settings", params: { profile: 1 } }'
+          )
+            q-item-section.q-pa-sm
+              q-btn(
+                flat,
+                icon='settings',
+                label='Settings',
+                dense,
+                align='left',
+                font-size='10px'
+              )
+          q-separator(v-if='isLoggedIn')
+          q-separator
           q-item(v-if='isLoggedIn')
-            q-item-section
-              | Signed in as
-              strong {{ profile.name }} {{ profile.surname }}
+            q-item-section.q-pa-lg
+              q-btn.logout-btn(
+                flat,
+                label='Sign Out',
+                dense,
+                font-size='10px',
+                @click='logout()'
+              )
+
+q-toolbar.row.q-py-sm.q-px-md.bg-grey-1(
```

Comment:
Let's try to minimize code duplication. I suspect doing the screen size check per component (discover button, account, search bar) rather than doing it for the entire toolbar would reduce the number of duplicated code a lot. This change request is an important one, because things can get very messy if we don't keep the code clean

Reply:
Hey Jacques, thanks for the reply. I'll take more time playing around with it and reduce the code lines